### ISSUE TITLE
Add support for replacing configured nodes during block device creation

### ISSUE THREAD

**Issue by vbellur** [gluster/gluster-block#11]
#gluster-block create <volume-name/device-name> ha <count> <host1>,<host2>,...<hostN> is the current CLI for creating a block device. 

If any of the hosts 1..N are to be replaced, gluster-block would need to provide an interface to perform that operation.

**Comment by vbellur** [comment]
Notes from an early discussion between @pranithk and @pkalever:

Granularity:

Since all the block operations (create/delete/list/info) are volume dependent, we intend supporting the migration functionality at the granularity of a volume. For replacing  a node all volumes that are part of the node would need to be migrated. Providing migration at the granularity of a volume can be helpful in scenarios where replace brick is performed.

Workflow:

1. Stop all targets from given volume gracefully. 

2. On the destination node we read all records from the metadata directory of the volume and create the targets freshly one-by-one. For every target recreation we do take lock and perform unlock once that individual target creation is done. 

3. Once the replacement is successful all the clients of old node have to
logout of the old portals and login to new portals.

**Comment by vbellur** [comment]
Proposed CLI for the operation:

#gluster-block modify <volname>  replace-node <old-node> <new-node>

**Comment by pkalever** [comment]
Duplicate of #4

**Comment by pkalever** [comment]
Please feel free to reopen the issue again. If you feel this cannot be addressed in #4
